django_x509/base/models.py

```diff
@@ -194,6 +196,20 @@ def pkey(self):
         if self.private_key:
             return crypto.load_privatekey(crypto.FILETYPE_PEM, self.private_key)
 
+    def _validate_pem(self):
+        """
+        (internal use only)
+        validates certificate and private key
+        """
+        try:
+            crypto.load_certificate(crypto.FILETYPE_PEM,self.certificate)
+        except OpenSSL.crypto.Error as e:
+            raise ValidationError({'certificate': _('OpenSSL error: {0}'.format(e.args[0]))})
+        try:
+           crypto.load_privatekey(crypto.FILETYPE_PEM, self.private_key)
+        except OpenSSL.crypto.Error as e:
+            raise ValidationError({'private_key': _('OpenSSL error: {0}'.format(e.args[0]))})
```

Comment:
great! Try to improve this method even more, see the following example code:

```python
errors = []
for field in ['certificate', 'private_key']:
    try:
        crypto.load_certificate(crypto.FILETYPE_PEM, getattr(self, field))
    except OpenSSL.crypto.Error as e:
        errors.append(ValidationError({
            field: _('OpenSSL error: {0}'.format(e.args[0]))}
        ))
if errors:
    raise ValidationError(errors)
```

See the [django docs about raising multiple errors](https://docs.djangoproject.com/en/1.11/ref/forms/validation/#raising-multiple-errors) to understand what the proposed code does.